### ISSUE TITLE
Sorting of created id column from rolling is awkward

### ISSUE THREAD

**Issue by TKlerx** [blue-yonder/tsfresh#699]
Hi there,

it has been some time since last working with tsfresh and I am impressed what you implemented in the mean time.
I am using the rolling feature and the newly created id column is not sorted as it should be.

1. Your operating system: Win10x64
2. The version of *tsfresh* that you are using: 0.16.0

Here is a minimal example:
```
import pandas as pd
from tsfresh.utilities.dataframe_functions import roll_time_series
from tsfresh import extract_features
df = pd.DataFrame({
   "initial_id": [1]*12,
   "time": [1, 2, 3, 4,5,6,7,8,9,10,20,30],
   "x": [1, 2, 3, 4,5,6,7,8,9,10,20,30]
})
window_size = 2
df_rolled = roll_time_series(df, column_id="id", column_sort="time", min_timeshift=window_size-1, max_timeshift=window_size-1)
df_rolled = df_rolled.drop('initial_id', axis = 1)
df_features = extract_features(df_rolled, column_id="id", column_sort="time", n_jobs = 8)
```
After sorting, the first id is `id=1,timeshift=10` and not `id=1,timeshift=1`.

I guess the issue is that the sort column is added as a string and if it is an integer, a '1' is not padded with some leading 0s.
This is not a real bug, but it is somehow awkward, because 10 is greater than 2 when treating 10 and 2 as integers instead of strings.

**Comment by nils-braun** [comment]
Welcome back, @TKlerx :-)

That is a very valid point. Unfortunately, we have to store multiple information (the old id and the timeshift) in a single column.
Would it be better if the column is a tuple (old id, timeshift) with the numerical values still present? We somehow would loose the "interpretability" a bit though...

**Comment by TKlerx** [comment]
Wow, that was a quick answer. I found that my example was not sufficient and wanted to edit a bit.
Actually, the problem arises when passing `df_rolled` to `extract_features` because I need to get rid of the old id and time, but only use the new id (id=<old_id>, timeshift) of type string.

I thought about a bigger change to be able to pass more than one sorting column to the `extract_features` method.

**Comment by nils-braun** [comment]
Making the sorting handle more than one columns is easy, the problem is that we would need to have the "column_id" handle more than one column.
In principle, that is not so hard - in practice it is of course all over the place.

Thanks for the heads-up on the memory consumption.
If you want, I can have a look into this - but I am happy for contributions :-)

**Comment by TKlerx** [comment]
My first workaround for the sorting is to transform the time column to string before and prepend leading 0s. But this does not solve the memory issue.

I will try a PR at least. Not sure whether I will get this working because I do not know all the dependencies. Will give you an update most probably next week after the long weekend.
Thanks for the quick replies!

**Comment by nils-braun** [comment]
Nice! Thanks for looking into this.

* The sorting issue could be fixed by converting the current string into a tuple (id, timeshift). That would also reduce the memory (but is still not as efficient as two columns).
That could be a first workaround.
* second possibility would be of course to add two new columns in the rolling function (instead of a merged one). But then all public-facing functions (extract_features, select_features, ...) would need to handle multiple columns as "column_id". It is definitely nice to have - but I am not sure if you want to implement all of that already now :-)

**Comment by TKlerx** [comment]
Yeah, I just forked again and checked the code. The first thing would be quite doable (I guess) but changing all the functions that take a `column_sort` is a big API change that I could not do (most probably).
Will work on the first one for now.

**Comment by TKlerx** [comment]
Will close this as it is corrected in #700
